k2/python/k2/rnnt_decode.py

```diff
@@ -264,4 +277,14 @@ def format_output(
             for name, value in src.named_non_tensor_attr():
                 setattr(fsa, name, value)
 
+        if log_probs is not None:
+            # Make fsa.scores tracked by autograd
+            # to make the whole decoding process differentiable.
+            scores_tracked_by_autograd = torch.index_select(
+                log_probs.reshape(-1), 0, arc_map_token)
+            final_arc_index = torch.where(fsa.arcs.values()[:, 2] == -1)
+            scores_tracked_by_autograd[final_arc_index] *= 0
```

Comment:
See the comment above, if you add `-1` to `arc_map_token` and use `k2.index_select` you can avoid line 285-286.